treeherder/perf/migrations/0012_rename_summary_last_updated.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.16 on 2019-02-20 15:02
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```

Comment:
Each time a prod push is made, the Heroku buildpack compile the `bin/pre_deploy` script in this repository is run, immediately prior to the new code being pushed out to the newly spun up dynos. It is in this script that the `./manage.py migrate` command is run, which will apply this migration.

This means that for X amount of time (where X is ~10-20 seconds + the time it takes for the migration to run) the schema in production will be updated, but the old code still being run by the existing dynos. This will cause API errors (HTTP 500s) and/or background job exceptions when Django can't find the fields it's expecting, until the new code is deployed. (This is not unique to Heroku or Django, in any production system a decision has to be made as to when to run the migrate steps relative to the code push, given that it's the same database being used before/after.)

The `performance_alert_summary` table is pretty small yeah? Also I believe rename operations like this are likely ~instant given I think MySQL just changes the metadata and not the table itself. As such I'm guessing this migration won't take more than a couple of seconds?

I'm presuming<30 seconds of errors are acceptable for API consumers, and the background perfherder tasks are all set to retry? I'll leave it to you to confirm this.

Reply:
(Edited to remove stray part sentence)

Reply:
> I'm presuming<30 seconds of errors are acceptable for API consumers, and the background perfherder tasks are all set to retry? I'll leave it to you to confirm this. (If not, then a set of backwards compatible migrations needs to be planned out instead, where at all the interim states the schema and old/new code are interchangeable)

I only see one background Perfherder task and that is `generate_perf_alerts`. I see that was already wrapped inside `retryable_task` decorator, with a max of 10 attempts.